### PR TITLE
[et][client-build] skip uploading by default, don't override files from CI

### PR DESCRIPTION
**Pull request by esamelson**
# Why

Incident in #platform-ios . Whenever the iOS client GH action job runs, it overwrites the production Exponent-2.16.0.tar.gz in S3 and users get it.

# How

- Skip uploading to S3 by default unless we are intentionally releasing. There is no reason to do this otherwise and it's too easy to overwrite files used in production.
- Change the default to never overwrite files from CI jobs. Again, too easy to accidentally screw things up in production.

# Test Plan

Need to ensure that the iOS client GH action workflow with this change does not upload anything to S3.

### CODE REVIEW

**Comment by ide** [comment]
Do you see a need to separate the upload & release steps? That is, should there be a `--upload` flag that replaces `--skipUpload`?

**Review by ide — APPROVED** [review]
Approving since this looks strictly better than what we had before but adding an `--upload` flag might be useful if we want to intentionally separate uploading from releasing.

**Comment by esamelson** [comment]
I don't think we'll ever want to upload without also writing to the versions endpoint (which is what releasing does) because that just seems confusing. But we can see what @tsapeta says

File: tools/expotools/src/commands/ClientBuild.ts [hunk 1]
```diff
@@ -100,7 +92,8 @@ async function askToRecreateSimulatorBuildAsync(): Promise<boolean> {
 
 async function askToOverrideBuildAsync(): Promise<boolean> {
   if (process.env.CI) {
-    return true;
+    // we should never override anything in CI, too easy to accidentally mess something up in prod
+    return false;
```
Comment (tsapeta) [inline tools/expotools/src/commands/ClientBuild.ts:96]: This is probably not correct — imho we do want to override builds on the CI. If we're during the release process (and let's say version `2.17.0` hasn't been released to production yet), we might want to override the simulator build (only on staging) without changing the version number (which would require additional manual work).

**Review by tsapeta — COMMENTED** [review]
Except `skipUpload` change, I think it does solve one problem but introduces another one (see comment below).

To make it right, we should add file checksum or git head hash to the filename of the tarball, as we do for shell app builds already and maybe add an expiration rule to S3 so that tarballs are removed from the bucket after a year or sth.

**Review by tsapeta — APPROVED** [review]
I'm approving this besides my concerns with overriding builds by CI and going to merge this as a temporary solution.

**Pull request merged**